### PR TITLE
Updated summary plugin to latest version

### LINKED ISSUE #283: Summary feature broke for 2020 posts
Pelican Summary works for all posts, except 2020 posts from the last update. 
Needs a hotfix.

### PR DESCRIPTION
**Pull request by ritesh-pandey**
* **Please check if the PR fulfills these requirements**

- [x] Screenshots for changes (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
#283 

* **What is the new behavior (if this is a feature change)?**
Fixes #283 
image

* **Other information**:

### CODE REVIEW

**Comment by ritesh-pandey** [comment]
What tests/commands are executed in Netlify?

On Ubuntu 20.04, default `virtualenv` was created with `Python3`. No such error occurred.
I think we use `Python2.7` in production. But I did not find any document which stated the tech stack.

It would have been easier if I had known this beforehand. Probably we should include it in contributing guideline.

**Comment by vipulgupta2048** [comment]
Thank your for the PR @ritesh-pandey, @palnabarun can help you with that answer

**Comment by palnabarun** [comment]
@ritesh-pandey Hi, we are using Python 3 as much as possible. So, the `markdown` package version pinning can be dropped.

Also, I found that most of the changes in this PR are from https://github.com/getpelican/pelican-plugins/pull/1211. Can you add a reference to the same in https://github.com/pythonindia/inpycon-blog/pull/286/commits/0fe2fe746421629c62da78ab83b833e5bd1317c4? Giving proper attribution is very important.

I have updated the build tooling in https://github.com/pythonindia/inpycon-blog/pull/288. Once, that is merged, the build will automatically install the dependencies and build the website.

**Comment by ritesh-pandey** [comment]
> So, the markdown package version pinning can be dropped.

At most places I have observed that requirement file use a fixed package version. It ensures that all development platforms are in absolute sync. They say "**relying on `latest`/`master` may introduce breaking changes in build process which are difficult to identify**". I don't come from Python background. Can you please tell me why we are going non-pinned approach?

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
+beautifulsoup4==4.9.1
```
Comment (ananyo2012) [inline requirements.txt:1]: Let's remove this requirement once #288 is merged. That will take care take care of installing plugin dependencies with the makefile target

Comment (ananyo2012) [reply requirements.txt:1]: Or maybe use `-r plugins/summary/requirements.txt` ? @palnabarun Thoughts ?

Comment (palnabarun) [reply requirements.txt:1]: That's what the `make requirements` does.

**Comment by palnabarun** [comment]
> They say "relying on latest/master may introduce breaking changes in build process which are difficult to identify". I don't come from Python background. Can you please tell me why we are going non-pinned approach?

@ritesh-pandey I should have been clearer with my statement. Version pinning is certainly better. I meant to say that we should not pin to the version which supports Python2 but to a version that is latest as well as stable.

File: requirements.txt (hunk 1 above)
Comment (ritesh-pandey) [reply requirements.txt:1]: To summarize, the upcoming build process will install dependencies from `requirements` file from main app and plugins. So, I should not remove the `requirements` file from plugin directory. Did I understand it correctly?

**Comment by palnabarun** [comment]
@ritesh-pandey Can you please rebase your changes against `master`?

**Comment by ritesh-pandey** [comment]
Does it look good now?

**Comment by ananyo2012** [comment]
@palnabarun Seems like #288 didn't fix the python version in Netlify. Can you check how to set the python version in netlify.toml ?

**Comment by vipulgupta2048** [comment]
@palnabarun ping

**Comment by vipulgupta2048** [comment]
@palnabarun @ananyo2012 @ritesh-pandey Any updated status on this?

**Comment by palnabarun** [comment]
Since we are now using the metadata field directly, do we need this change anymore? @vipulgupta2048 @Sc0rpi0n101

**Comment by Sc0rpi0n101** [comment]
I don't think we even need the summary plugin anymore since it's not working properly anyway. I'll test to see removing it doesn't break anything and make a PR if it doesn't.

In the meantime, I think this can be closed.

**Comment by vipulgupta2048** [comment]
Makes sense, closing..

**Pull request closed without merging**